compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.internal
+
+import androidx.compose.ui.platform.ClipEntry
+import androidx.compose.ui.text.AnnotatedString
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.StringSelection
+import java.awt.datatransfer.UnsupportedFlavorException
+import java.io.IOException
+
+internal actual suspend fun ClipEntry.readText(): String? {
+    return try {
+        transferable.getTransferData(DataFlavor.stringFlavor) as String?
+    } catch (_: UnsupportedFlavorException) {
+        null
+    } catch (_: IllegalStateException) {
+        null
+    } catch (_: IOException) {
+        null
+    }
+}
+
+internal actual suspend fun ClipEntry.readAnnotatedString(): AnnotatedString? {
+    return readText()?.let { return AnnotatedString(it) }
+}
+
+internal actual fun AnnotatedString?.toClipEntry(): ClipEntry? {
+    if (this == null) return null
+    val transferable = StringSelection(this.text)
+    return ClipEntry(transferable)
+}
+
+internal actual fun ClipEntry?.hasText(): Boolean {
+    return try {
+        (this?.transferable?.getTransferData(DataFlavor.stringFlavor) as? String)?.isNotEmpty() == true
```

Comment:
1. Retrieving the data could be a long process. Why not just return `isDataFlavorSupported(DataFlavor.string)`? I don't think there's anyway a guarantee that if `hasText` returns `true`, then `readText` will return a non-null value (the process that is the source of the clipboard data could've died in between).
2. I would suggest a simple `if (this == null) return false` instead of a long string of `?.`.
3. `!string.isNullOrEmpty` looks better than `string?.isNotEmpty() == true`? But also, why do we return `false` for an empty string?